### DISCUSSION TITLE
outdated dag code on airflow webserver

### DISCUSSION THREAD

**Discussion by enniomorricone123** [apache/airflow#21580] [General]
### Apache Airflow version

2.2.3 (latest released)

### What happened

Since airflow > 2.0.0 [core] store_dag_code has been removed (see https://github.com/apache/airflow/pull/16342). However, this created huge problem because now the webserver (UI) shows very outdated dag code. Only scheduler has serialized new dag, but the source code of dag is not updated in DB therefore not showing in dag code view of UI. 

Is there a way to show most updated dag code in webserver?

Can we bring back the store_dag_code optional to allow webserver to read dag from the local files? or is there a way to update the DB (dag.source_code)?

### Operating System

linux

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
This is not how it behaves. Airlfow scheduled will store both serialzied form and the DAG code in the database. So it should be in sync - whatever scheduler sees localy is stored in the DB. 

There might be some bugs (of course) and some external settings in your configuration that might prevent this from happening, but this should be tracked down on your side. 

If you can see some evidences, provide logs and explain your configuration when you see outdated code  in the UI, please open an issue describing it, but the reason is not lack of "store_dag_code" parameter.

**Comment by potiuk** [comment]
converted it into discussion as this is not an issue
